src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIAnalysesOutputsService.java

```diff
@@ -0,0 +1,172 @@
+		this.analysisSubmissionService = analysisSubmissionService;
+		this.workflowsService = workflowsService;
+		this.userService = userService;
+		this.analysisOutputFileDownloadManager = analysisOutputFileDownloadManager;
+	}
+
+	/**
+	 * Get all the shared single sample analysis outputs for the project
+	 *
+	 * @param projectId {@link ca.corefacility.bioinformatics.irida.model.project.Project} id
+	 * @return a list of filtered {@link ProjectSampleAnalysisOutputInfo} single sample analysis outputs
+	 */
+	public List<ProjectSampleAnalysisOutputInfo> getSharedSingleSampleOutputs(Long projectId) {
+		List<ProjectSampleAnalysisOutputInfo> projectSampleAnalysisOutputInfos = analysisSubmissionService.getAllAnalysisOutputInfoSharedWithProject(
+				projectId);
+		return getSingleSampleAnalysisOutputsInfo(projectSampleAnalysisOutputInfos);
+	}
+
+	/**
+	 * Get all the automated single sample analysis outputs for the project
+	 *
+	 * @param projectId {@link ca.corefacility.bioinformatics.irida.model.project.Project} id
+	 * @return a list of filtered {@link ProjectSampleAnalysisOutputInfo} single sample analysis outputs
+	 */
+	public List<ProjectSampleAnalysisOutputInfo> getAutomatedSingleSampleOutputs(Long projectId) {
+		List<ProjectSampleAnalysisOutputInfo> projectSampleAnalysisOutputInfos = analysisSubmissionService.getAllAutomatedAnalysisOutputInfoForAProject(
+				projectId);
+		return getSingleSampleAnalysisOutputsInfo(projectSampleAnalysisOutputInfos);
+	}
+
+	/**
+	 * Get all the automated single sample analysis outputs for the project
+	 *
+	 * @param principal Currently logged in user.
+	 * @return a list of filtered {@link ProjectSampleAnalysisOutputInfo} single sample analysis outputs for the user
+	 */
+	public List<ProjectSampleAnalysisOutputInfo> getUserSingleSampleOutputs(Principal principal) {
+		User user = userService.getUserByUsername(principal.getName());
+		List<ProjectSampleAnalysisOutputInfo> userProjectSampleAnalysisOutputInfos = getSingleSampleAnalysisOutputsInfo(
+				analysisSubmissionService.getAllUserAnalysisOutputInfo(user));
+
+		// Need to set the user information to the currently logged in user
+		userProjectSampleAnalysisOutputInfos.forEach(singleSampleAnalysisOutput -> {
+			singleSampleAnalysisOutput.setUserId(user.getId());
+			singleSampleAnalysisOutput.setUserFirstName(user.getFirstName());
+			singleSampleAnalysisOutput.setUserLastName(user.getLastName());
+		});
+		return userProjectSampleAnalysisOutputInfos;
+	}
+
+	/**
+	 * Prepare the download of multiple {@link AnalysisOutputFile} by adding them to a selection.
+	 *
+	 * @param outputs  Info for {@link AnalysisOutputFile} to download
+	 * @param response {@link HttpServletResponse}
+	 */
+	public void prepareAnalysisOutputsSelectionDownload(List<ProjectSampleAnalysisOutputInfo> outputs,
+			HttpServletResponse response) {
+		analysisOutputFileDownloadManager.setSelection(outputs);
+		response.setStatus(HttpServletResponse.SC_CREATED);
```

Comment:
Why are you setting the status of the response in the service?  This is really one of the only things that should be done in the controller.  If the service does not fail then set a `ResponseEntity` with the status of created.

Reply:
This was something that was there before and I just reused. I can update it :)

Reply:
Updated in c6f2f6cfa2f5b74494bf05a58d24fd7fb08c35fb